XiEditor/Search.swift

```diff
@@ -0,0 +1,208 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+import Swift
+
+class FindViewController: NSViewController, NSTextFieldDelegate {
+
+    @IBOutlet weak var searchField: NSSearchField!
+    @IBOutlet weak var navigationButtons: NSSegmentedControl!
+    @IBOutlet weak var doneButton: NSButton!
+    @IBOutlet weak var ignoreCaseMenuItem: NSMenuItem!
+    @IBOutlet weak var wrapAroundMenuItem: NSMenuItem!
+
+    var optionIgnoreCase: Bool = true
+    var optionWrapAround: Bool = true
+
+    var editViewController: EditViewController!
+
+    override func viewDidLoad() {
+        // add recent searches menu items
+        let menu = searchField.searchMenuTemplate!
+
+        menu.addItem(NSMenuItem.separator())
+
+        let recentTitle = NSMenuItem(title: "Recent Searches", action: nil, keyEquivalent: "")
+        recentTitle.tag = Int(NSSearchFieldRecentsTitleMenuItemTag)
+        menu.addItem(recentTitle)
+
+        let recentItem = NSMenuItem(title: "", action: nil, keyEquivalent: "")
+        recentItem.tag = Int(NSSearchFieldRecentsMenuItemTag)
+        menu.addItem(recentItem)
+
+        menu.addItem(NSMenuItem.separator())
+
+        let recentClear = NSMenuItem(title: "Clear Recent Searches", action: nil, keyEquivalent: "")
+        recentClear.tag = Int(NSSearchFieldClearRecentsMenuItemTag)
+        menu.addItem(recentClear)
+    }
+
+    override func validateMenuItem(_ menuItem: NSMenuItem) -> Bool {
+        return true
+    }
+
+    @IBAction func selectIgnoreCaseMenuItem(_ sender: Any) {
+        if let menuItem = sender as? NSMenuItem {
+            optionIgnoreCase = !optionIgnoreCase
+            if optionIgnoreCase {
+                menuItem.state = NSOnState
+            } else {
+                menuItem.state = NSOffState
+            }
+        }
+
+        editViewController.find(searchField.stringValue,
+                                case_sensitive: !optionIgnoreCase,
+                                wrap_around: optionWrapAround)
+    }
+    
+    @IBAction func selectWrapAroundMenuItem(_ sender: Any) {
+        if let menuItem = sender as? NSMenuItem {
+            optionWrapAround = !optionWrapAround
+            if optionWrapAround {
+                menuItem.state = NSOnState
+            } else {
+                menuItem.state = NSOffState
+            }
+        }
+    }
+    
+    @IBAction func clickSegmentControl(_ sender: NSSegmentedControl) {
```

Comment:
nit: I would call this `segmentControlAction`, or `findButtonsAction` or `findControlAction`.